syft/pkg/java_metadata.go

```diff
@@ -20,7 +20,7 @@ var jenkinsPluginPomPropertiesGroupIDs = []string{
 
 // JavaMetadata encapsulates all Java ecosystem metadata for a package as well as an (optional) parent relationship.
 type JavaMetadata struct {
-	VirtualPath    string         `json:"virtualPath" cyclonedx:"virtualPath"` // we need to include the virtual path in cyclonedx documents to prevent deduplication of jars within jars
```

Comment:
I don't think we can ignore this since it can help distinguish between two logically separate packages.

Example:
- We have a jenkins plugin that is tested where they are duplicates EXCEPT for the virtual path they are installed. For the first one `VirtualPath: /java/example-jenkins-plugin.hpi`. For the second one `VirtualPath: /java/example-jenkins-plugin.hpi:WEB-INF/lib/example-jenkins-plugin.jar` <-- This is what allows syft to detect separate packages nested within their parents with the same name.

I think ignoring the Locations list is ok as far as duplicates are concerned, but this field shows is similarly named packages are located within each other or at separate locations.